flang/test/Integration/OpenMP/workshare-axpy.f90

```diff
@@ -13,7 +13,7 @@ subroutine sb1(a, x, y, z)
   integer :: a
   integer :: x(:)
   integer :: y(:)
-  integer :: z(:)
+  integer, allocatable :: z(:)
```

Comment:
Is this change required?

Reply:
Yes, this change is required, if you want to keep testing the fact that the `fir.allocmem` and `fir.freemem` are placed in the `omp.single` regions. Those operations appear due to the optimized bufferization not being able to optimize the assignment without a runtime call, which requires the RHS to be computed into a temporary array.

With this patch the line 18 statement will be optimized such that there is no runtime call at all, and hence no temporary array allocation/deallocation.

Reply:
I see. thank you very much. I mistakenly thought the optimization was happening already.